### DISCUSSION TITLE
Use variable in overlay use command

### DISCUSSION THREAD

**Discussion by Tom-Boscher** [nushell/nushell#12206] [Q&A] [answered]
I have this command `let venv_path = $env.ZED_WORKTREE_ROOT + '/.venv/bin/activate.nu' ; echo $venv_path` which output : `/Users/tom/Desktop/codedex/.venv/bin/activate.nu`. So my variable `venv_path` is the string `/Users/tom/Desktop/codedex/.venv/bin/activate.nu`.
When I tried to use it with this command `let venv_path = $env.ZED_WORKTREE_ROOT + '/.venv/bin/activate.nu' ; overlay use $venv_path ; python $env.ZED_FILE` I encounter this error: `Value is not a parse-time constant`

Is it possible to use a variable in `overlay use` command?

**Comment by fdncred** [comment]
I'm wondering if you change it from a `let venv_path` to a `const venv_path` if it would work. Have you tried that?

**Reply by Tom-Boscher** [reply]
I indeed tried using `const` instead of `env` but I encountered the same error
```
  │  1 │ const venv_path = $env.ZED_WORKTREE_ROOT + '/.venv/bin/activate.nu' ; overlay use $venv_path ; python $env.ZED_FILE
  │    ·                                  ─┬-
  │    ·                                     ╰── Value is not a parse-time constant
```

**Reply by ghost** [reply, marked as answer]
you currently cant access mutable variables (including `$env` in a const assignment).  
and the original issue is that `overlay use` only accepts constants as argument.

a workaround is to first generate a file (in `env.nu`):
```nu
$"use '($env.JED_WORKTREE_ROOT)/.venv/bin/activate.nu' *" | save -r "~/.tmp.nu"
```
and then load it later (in `config.nu`):
```nu
overlay use ~/.tmp.nu
```

this has to be done at seperate times, since nu checks if the `overlay use` file exists at file load time and not when the command is called.  
but if you are running both commands by hand in the REPL you might get away with it.

my solution to python venvs is this: <https://github.com/Jan9103/nushell_python/blob/main/pyvenv.nu>

**Reply by Tom-Boscher** [reply]
@Jan9103 Ty, it's a snippet task I used in Zed editor so my goal was to activate the venv when launch my python file using this snippet.
I will create a file then.
